reposerver/repository/repository.go

```diff
@@ -1294,27 +1294,39 @@ func getPluginEnvs(envVars *v1alpha1.Env, q *apiclient.ManifestRequest, creds gi
 
 	if q.ApplicationSource.Plugin != nil {
 		pluginEnv := q.ApplicationSource.Plugin.Env
-		for i, j := range pluginEnv {
-			pluginEnv[i].Value = parsedEnv.Envsubst(j.Value)
+		for _, entry := range pluginEnv {
+			newValue := parsedEnv.Envsubst(entry.Value)
+			if legacy {
+				env = append(env, fmt.Sprintf("%s=%s", entry.Name, newValue))
+			}
+			env = append(env, fmt.Sprintf("ENV_%s=%s", entry.Name, newValue))
```

Comment:
@pierrecregut sorry for the 2nd review, but I'm wondering if you would change this to `ARGOCD_ENV_`. I just suddenly became worried that this might have collisions.

Reply:
Done.